tombstones.go

```diff
@@ -0,0 +1,80 @@
+package zoekt
+
+import (
+	"bufio"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"runtime"
+	"syscall"
+)
+
+// TombstoneFileName if present in IndexDir will create *.rip files containing
+// tombstones instead of deleting compound shards.
+const TombstoneFileName = "RIP"
+
+// SetTombstone idempotently adds repoName to the .rip file of the shard at
+// shardPath. It does not validate whether the repository is actually contained
+// in the shard.
+func SetTombstone(shardPath string, repoName string) error {
+	ts, err := LoadTombstones(shardPath)
+	if err != nil {
+		return err
+	}
+
+	ts[repoName] = struct{}{}
+
+	tmp, err := ioutil.TempFile(filepath.Dir(shardPath), filepath.Base(shardPath)+".*.tmp")
+	if err != nil {
+		return err
+	}
+	defer tmp.Close()
```

Comment:
also `defer os.Remove(tmp.Name())` so we clean up in the case of returning an error.